python/GafferImageTest/PremultiplyTest.py

```diff
@@ -0,0 +1,108 @@
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import os
+import unittest
+
+import IECore
+
+import Gaffer
+import GafferImage
+
+class PremultiplyTest( unittest.TestCase ) :
+
+	checkerFile = os.path.expandvars( "$GAFFER_ROOT/python/GafferTest/images/rgbOverChecker.100x100.exr" )
+
+	def testChannelDataHashes( self ) :
+		# Create a premult node and save the hash of a tile from each channel.
+		i = GafferImage.ImageReader()
+		i["fileName"].setValue( self.checkerFile )
+
+		premult = GafferImage.Premultiply()
+		premult["in"].setInput(i["out"])
+
+		h1 = premult["out"].channelData( "R", IECore.V2i( 0 ) ).hash()
+		h2 = premult["out"].channelData( "R", IECore.V2i( GafferImage.ImagePlug().tileSize() ) ).hash()
+		self.assertNotEqual( h1, h2 )
+
+		# Test that two tiles within the same image have the same hash when disabled.
+		premult["enabled"].setValue(False)
+		h1 = premult["out"].channelData( "R", IECore.V2i( 0 ) ).hash()
+		h2 = premult["out"].channelData( "R", IECore.V2i( GafferImage.ImagePlug().tileSize() ) ).hash()
+		self.assertNotEqual( h1, h2 )
+
+	def testByChannel( self ) :
+		# Test that changing the channel to multiply by changes the hash
+		i = GafferImage.ImageReader()
+		i["fileName"].setValue( self.checkerFile )
+
+		premult = GafferImage.Premultiply()
+		premult["in"].setInput(i["out"])
+
+		premult["byChannel"].setValue("R")
+		h1 = premult["out"].channelData( "R", IECore.V2i( 0 ) ).hash()
+
+		premult["byChannel"].setValue("B")
+		h2 = premult["out"].channelData( "R", IECore.V2i( 0 ) ).hash()
+		self.assertNotEqual( h1, h2 )
+
+	def testEnableBehaviour( self ) :
+
+		g = GafferImage.Premultiply()
+		self.assertTrue( g.enabledPlug().isSame( g["enabled"] ) )
+		self.assertTrue( g.correspondingInput( g["out"] ).isSame( g["in"] ) )
+		self.assertEqual( g.correspondingInput( g["in"] ), None )
+		self.assertEqual( g.correspondingInput( g["enabled"] ), None )
+		self.assertEqual( g.correspondingInput( g["byChannel"] ), None )
+
+	def testPassThrough( self ) :
+
+		i = GafferImage.ImageReader()
+		i["fileName"].setValue( self.checkerFile )
+
+		g = GafferImage.Premultiply()
+		g["in"].setInput( i["out"] )
+
+		self.assertEqual( i["out"]["format"].hash(), g["out"]["format"].hash() )
+		self.assertEqual( i["out"]["dataWindow"].hash(), g["out"]["dataWindow"].hash() )
+		self.assertEqual( i["out"]["metadata"].hash(), g["out"]["metadata"].hash() )
+		self.assertEqual( i["out"]["channelNames"].hash(), g["out"]["channelNames"].hash() )
+
+		self.assertEqual( i["out"]["format"].getValue(), g["out"]["format"].getValue() )
+		self.assertEqual( i["out"]["dataWindow"].getValue(), g["out"]["dataWindow"].getValue() )
+		self.assertEqual( i["out"]["metadata"].getValue(), g["out"]["metadata"].getValue() )
+		self.assertEqual( i["out"]["channelNames"].getValue(), g["out"]["channelNames"].getValue() )
```

Comment:
I think we really need a test here that actually deals in pixel values. How about one that uses a Constant node to generate known values, and then asserts that the values coming out of the premultiply are as expected?